src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/TracePoint.kt

```diff
@@ -182,28 +193,45 @@ internal class MethodCallTracePoint(
     private var ownerName: String? = null
 
     val wasSuspended get() = (returnedValue == ReturnedValueResult.CoroutineSuspended)
+    private val paramNames: List<String> by lazy {
+        val methods = Class.forName(className.replace('/', '.')).methods
+        val method = methods.firstOrNull { Type.getMethodDescriptor(it) == descriptor }
+        method?.kotlinFunction?.parameters?.map { it.name ?: "" } ?: emptyList()
+    }
 
     override fun toStringCompact(): String = StringBuilder().apply {
-        if (ownerName != null)
-            append("$ownerName.")
-        append("$methodName(")
-        val parameters = parameters
-        if (parameters != null) {
-            append(parameters.joinToString(", "))
-        }
-        append(")")
-        val returnedValue = returnedValue
-        if (returnedValue is ReturnedValueResult.ValueResult) {
-            append(": ${returnedValue.valueRepresentation}")
-        } else if (returnedValue is ReturnedValueResult.CoroutineSuspended) {
-            append(": COROUTINE_SUSPENDED")
-        } else if (thrownException != null && thrownException != ThreadAbortedError) {
-            append(": threw ${thrownException!!.javaClass.simpleName}")
+        // If thread creation site
+        if (descriptor == THREAD_FUN_DESCRIPTOR && className == THREAD_CLASS_NAME) {
+            append("thread")
+            val params = parameters?.checkDefault(paramNames, threadDefaultValues) ?: emptyList()
+            if (!params.isEmpty()) {
+                append("(")
+                append(params.joinToString(", "))
+                append(")")
+            }
+            append(" { ... } ")
```

Comment:
Please extract `thread` function printing logic into separate function. 
Then also replace `if` with `when` (as we might need to add more special cases in the future).